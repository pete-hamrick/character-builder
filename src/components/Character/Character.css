.character {
  display: flex;
  flex-direction: column;
}

section {
  background-position: center;
}
