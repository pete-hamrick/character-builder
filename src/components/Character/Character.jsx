import React from 'react'

export default function Character() {
  return (
    <section>
      <h1>Character Display</h1>
    </section>
  )
}
