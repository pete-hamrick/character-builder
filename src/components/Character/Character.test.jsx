import { render, screen } from '@testing-library/react'
import Character from './Character'
import React from 'react'

it('should display the Character component', () => {
  const { container } = render(
    <Character hat="trucker" head="otter" shirt="fishing" bottom="kilt" />
  )

  const selectHatImage = screen.getByLabelText('trucker')
  const selectHeadImage = screen.getByLabelText('otter')
  const selectShirtImage = screen.getByLabelText('fishing')
  const selectBottomImage = screen.getByLabelText('kilt')

  expect(selectHatImage).toBeInTheDocument()
  expect(selectHeadImage).toBeInTheDocument()
  expect(selectShirtImage).toBeInTheDocument()
  expect(selectBottomImage).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
