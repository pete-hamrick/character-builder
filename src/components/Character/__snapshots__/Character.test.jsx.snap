// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the Character component 1`] = `
<div>
  <article
    class="character"
  >
    <section
      aria-label="trucker"
      class="hat"
      style="background-image: url(./trucker-hat.png); height: 150px; background-size: contain; background-repeat: no-repeat;"
    />
    <section
      aria-label="otter"
      class="head"
      style="background-image: url(./otter-head.png); height: 150px; background-size: contain; background-repeat: no-repeat;"
    />
    <section
      aria-label="fishing"
      class="shirt"
      style="background-image: url(./fishing-shirt.png); height: 200px; background-size: contain; background-repeat: no-repeat;"
    />
    <section
      aria-label="kilt"
      class="bottom"
      style="background-image: url(./kilt-bottom.png); height: 230px; background-size: contain; background-repeat: no-repeat;"
    />
  </article>
</div>
`;
